main_libraryhelper.go

```diff
@@ -0,0 +1,46 @@
+package main
+
+import (
+	"fmt"
+	"syscall/js"
+
+	"github.com/AndreasSko/go-jwlm/wasm"
+)
+
+func mergeJs(this js.Value, inputs []js.Value) interface{} {
+	leftDbArr := inputs[0]
+	rightDbArr := inputs[1]
+	mergedDbName := inputs[2].String()
+	//mergedDbArr := make([]uint8, leftDbArr.Get("byteLength").Int())
+
+	leftBuf := make([]uint8, leftDbArr.Get("byteLength").Int())
+	rightBuf := make([]uint8, rightDbArr.Get("byteLength").Int())
+
+	js.CopyBytesToGo(leftBuf, leftDbArr)
+	js.CopyBytesToGo(rightBuf, rightDbArr)
+
+	mergedDb := wasm.Merge(leftBuf, rightBuf, mergedDbName)
+
+	//js.CopyBytesToJS(mergedDbArr, mergedDb)
+	fmt.Printf("Merged. Returning %d bytes\n", len(mergedDb))
+	mergedJsData := js.Global().Get("Uint8Array").New(len(mergedDb))
+	js.CopyBytesToJS(mergedJsData, mergedDb)
+	return mergedJsData
+
+}
+
+func registerCallbacks() {
+	js.Global().Set("mergeJs", js.FuncOf(mergeJs))
+}
+
+func main() {
+	//https://blog.twitch.tv/de-de/2019/04/10/go-memory-ballast-how-i-learnt-to-stop-worrying-and-love-the-heap-26c2462549a2/
+	ballast := make([]byte, 100<<20) //100MiB
```

Comment:
Interesting article, but I don‘t really see a reason to do it. Go is in general quite good with GC and the article talks about a specific use-case. In the case of go-jwlm the complete process should only take a few seconds (if not even less) anyway, so I don‘t think that GC really „slows down“ the program. But maybe you can explain your thought behind it a bit? 😊

Reply:
Yes, I agree. Was adding this at some point to keep it for performance tests later. Also do not expect a huge benefit. Will do some cleanup, fix the build annotations and add some tests anyway.